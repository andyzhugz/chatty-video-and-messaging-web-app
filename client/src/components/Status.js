import React, { useState, useEffect } from 'react';
import NaviBar from './NaviBar';
// import { getFeed, getStatus } from '../javascripts/status';
import '../stylesheets/Status.css';
import Avatar from '../images/AvatarCat.png';

const Status = () => {
  const [username, setUsername] = useState('');
    /**
    const [feed, setFeed] = useState(null);
    const [status, setStatus] = useState(null);

    const generateFeed = async() => {
        // const feed = getFeed();
        const feed = ['A', 'B', 'C'];
        feed.forEach((feed) => {
            feedArray.push(
                <div class="circle">{feed}</div>
            ); 
        });  
    };   
    */

   useEffect(() => {
    const loggedInUser = localStorage.getItem("curr_user");
    if (loggedInUser && loggedInUser !== "") {
      setUsername(loggedInUser);
    }
    else {
      window.open("/login","_self");
    }
  }, []);


  const displayStatus = async(type) => {
      // const status = getStatus();
      if (type === 'img') {
          const statusDiv = document.createElement('img');
          statusDiv.setAttribute('class', 'imageStatus');
          statusDiv.src = Avatar;
          document.getElementById('status').innerHTML = '';
          document.getElementById('status').appendChild(statusDiv);
      }
      else {
          const statusDiv = document.createElement('p');
          statusDiv.setAttribute('class', 'textStatus');
          statusDiv.innerText = 'Hello';
          document.getElementById('status').innerHTML = '';
          document.getElementById('status').appendChild(statusDiv);
      }
  }

  function sendStatus() {
      closepopWindow();
  }

  function popupWindow() {
      const popup = document.getElementById('popup1');
      popup.style.visibility = 'visible';
      const sendStatus = document.getElementById('sendStatus');
      sendStatus.style.visibility = 'hidden';
    }
  
  function closepopWindow() {
      const popup = document.getElementById('popup1');
      popup.style.visibility = 'hidden';
      const sendStatus = document.getElementById('sendStatus');
      sendStatus.style.visibility = 'hidden';
      const textarea = document.getElementById('textarea');
      textarea.value = '';
      textarea.style.visibility = 'hidden';
  }

  function displayTextArea() {
      const textarea = document.getElementById('textarea');
      textarea.style.visibility = 'visible';
      const sendBtn = document.getElementById('sendStatus');
      sendBtn.style.visibility = 'visible';
  }

  // On file upload (click the upload button)
  const onFileUpload = (selectedFile) => {
      // Create an object of formData
      const formData = new FormData();
      // Update the formData object
      formData.append('myFile', selectedFile, selectedFile.name);
      // Request made to the backend api
      // Send formData object
      //axios.post("api/uploadfile", formData);
  };

  function selectImage(event) {
      const selectedFile = event.target.files[0];
      if (selectedFile) {
          onFileUpload(selectedFile);
          // finally we should use the data retrieved from mongoDB in setMessage
          const src = URL.createObjectURL(selectedFile);
          // image
          if (
            selectedFile.name.endsWith('.png') ||
            selectedFile.name.endsWith('.jpg')
          ) {
            const imgDiv = `<img width="320" height="240" src=${src} alt="The picture is gone.">`;
            //setMessage(imgDiv);
          }
      }
  }

  const feedArray = [<div className="circle" key='0'><button id='btn0' className="statusLink" onClick={displayStatus.bind(this, 'img')}>A</button></div>, 
      <div className="circle" key='1'><button id='1' className="statusLink" onClick={displayStatus.bind(this, 'text')}>B</button></div>];

  return (
    <div>
      <NaviBar />
      <h1 align="center">Status</h1>
      <p align="center" onClick={() => popupWindow()}><button>Send New Status</button></p>
      <div className="mainProfile">
        <div className="outer" id='outer'>
          {feedArray}
        </div>
        <div id='status'>  </div>
        <div id="popup1" className="overlay">
        <div className="popup">
          <h2>New Status</h2>
          <a id="closepop" className="close" onClick={() => closepopWindow()}>Close</a>
          <p>
              <button id="textOption" onClick={() => displayTextArea()}>Text</button>
              <button id='imageOption'>
                  <label htmlFor="image-upload">Image</label>
                  <input
                      id="image-upload"
                      type="file"
                      onChange={selectImage}
                      accept="image/png, image/jpg"
                  />
              </button>
              <textarea id='textarea' rows="4" cols="40" style={{visibility: 'hidden'}}></textarea>
              <button id="sendStatus" onClick={() => sendStatus()}>
                  SEND
              </button>
          </p>
        </div>
      </div>
      </div>
    </div>
  );
};


export default Status;