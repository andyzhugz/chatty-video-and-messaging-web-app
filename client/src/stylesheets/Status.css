.outer {
    align-items: center;
    height: 80px;
    display: flex;
    flex-wrap: nowrap;
    overflow-x: auto;
    flex-wrap: wrap;
}

.circle {
    width: 60px;
    height: 60px;
    line-height: 60px;
    border-radius: 60%;
    font-size: 20px;
    color: #fff;
    text-align: center;
    background: #000;
    margin-left: 10px;
    margin-top: 3px;
  }

  .statusLink {
    background: none!important;
    border: none;
    padding: 0!important;
    /*optional*/
    font-family: arial, sans-serif;
    font-size: 20px;
    /*input has OS specific font-family*/
    color: #069;
    cursor: pointer;
  }

  
.imageStatus {
    width: 300px;
    height: 300px;
    margin: 25px 150px;
    border-radius: 1.5em;
    box-shadow: 0px 11px 35px 2px rgba(0, 0, 0, 0.14);
}

.textStatus {
    margin-top: 100px;
    text-align: center;
    font-family: arial, sans-serif;
    font-size: 30px;
}