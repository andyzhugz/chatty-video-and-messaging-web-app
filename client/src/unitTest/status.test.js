import React from 'react';
import ReactDOM from 'react-dom';
import Status from '../components/Status.js';
import { act } from 'react-dom/test-utils';

const renderer = require('react-test-renderer');

// snapshot testings
describe('Test snapshot', () => {
  test('Test Profile', () => {
    const component = renderer.create(<Status />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// functonal component functions tests
describe('Test functonal component functions', () => {
    let container;
    beforeEach(() => {
      container = document.createElement('div');
      document.body.appendChild(container);
    });
  
    afterEach(() => {
      document.body.removeChild(container);
      container = null;
    });
  
    test('Test function inside Status component', () => {
      act(() => {
            ReactDOM.render(<Status />, container);
      });
      const submitBtn = container.querySelector('#btn0');
      act(() => {
        fetch = jest.fn(() => Promise.resolve({ json: () => ({ message: 'success', data: [{ username: 'tester', password: 'tester' }] }) }));
        submitBtn.dispatchEvent(new MouseEvent('click', { bubbles: true }));
      });


    });

  });