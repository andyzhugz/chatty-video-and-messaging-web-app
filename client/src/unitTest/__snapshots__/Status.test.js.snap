// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Test Profile 1`] = `
<div>
  <div>
    <ul
      id="nav_ul"
    >
      <li
        className="nav_li"
      >
        <a
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li
        className="nav_li"
      >
        <a
          href="/main"
        >
          Messaging
        </a>
      </li>
      <li
        className="nav_li"
      >
        <a
          href="/status"
        >
          Status
        </a>
      </li>
    </ul>
  </div>
  <h1
    align="center"
  >
    Status
  </h1>
  <p
    align="center"
    onClick={[Function]}
  >
    <button>
      Send New Status
    </button>
  </p>
  <div
    className="mainProfile"
  >
    <div
      className="outer"
      id="outer"
    >
      <div
        className="circle"
      >
        <button
          className="statusLink"
          id="btn0"
          onClick={[Function]}
        >
          A
        </button>
      </div>
      <div
        className="circle"
      >
        <button
          className="statusLink"
          id="1"
          onClick={[Function]}
        >
          B
        </button>
      </div>
    </div>
    <div
      id="status"
    >
        
    </div>
    <div
      className="overlay"
      id="popup1"
    >
      <div
        className="popup"
      >
        <h2>
          New Status
        </h2>
        <a
          className="close"
          id="closepop"
          onClick={[Function]}
        >
          Close
        </a>
        <p>
          <button
            id="textOption"
            onClick={[Function]}
          >
            Text
          </button>
          <button
            id="imageOption"
          >
            <label
              htmlFor="image-upload"
            >
              Image
            </label>
            <input
              accept="image/png, image/jpg"
              id="image-upload"
              onChange={[Function]}
              type="file"
            />
          </button>
          <textarea
            cols="40"
            id="textarea"
            rows="4"
            style={
              Object {
                "visibility": "hidden",
              }
            }
          />
          <button
            id="sendStatus"
            onClick={[Function]}
          >
            SEND
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;
